worker-core/src/main/java/com/hpe/caf/worker/core/WorkerCore.java

```diff
@@ -396,7 +394,7 @@ public void complete(final String queueMsgId, final String queue, final TaskMess
                     // **** Normal Worker ****                    
                     // A worker with an input and output queue.
                     byte[] output = codec.serialise(responseMessage);
-                    workerQueue.publish(queueMsgId, output, targetQueue, Collections.emptyMap());
+                    workerQueue.publish(queueMsgId, output, targetQueue, Collections.emptyMap(), responseMessage.getPriority() == null ? 0 : responseMessage.getPriority());
```

Comment:
You're adding the priority here in the `complete()` method but not adding it in the `forward()` method.  Does that mean the priorities are getting lost when then Job Service is in the mix and the Job Tracking Worker is forwarding messages to the target worker?